galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/TestRunner.java

```diff
@@ -459,7 +459,7 @@ private void createEnvironment(TestClassWrapper testClassWrapper, TestRunManager
         try {
             if (this.runType == RunType.Test || this.runType == RunType.SharedEnvironmentBuild) {
                 try {
-                    updateStatus("building", null);
+                    updateStatus(TestRunLifecycleStatus.BUILDING, null);
```

Comment:
```
if (e instanceof FrameworkResourceUnavailableException) {
   this.resourcesUnavailable = true;
}
```
That code seems to appear lots of times. instanceof isn't a very OO thing to use in code. Better to have a method on the FrameworkException to ask `isResourcesUnavailable()` ?
(though I don't like the negative, I think we'lll have to live with it instead of `isResourcesAvailable()`)

Reply:
So just general question about best coding practice here, is it better to use more "positive" variables then?

Reply:
Have changed the variable name, however the logic has no tests behind it so apprehensive to change.

Reply:
Will open an issue for putting in unit tests and refactoring this.